pandas/core/frame.py

```diff
@@ -3648,17 +3648,34 @@ def transpose(self, *args, copy: bool = False) -> DataFrame:
             and dtypes
             and isinstance(dtypes[0], ExtensionDtype)
         ):
+            new_values: list
             if isinstance(dtypes[0], BaseMaskedDtype):
                 # We have masked arrays with the same dtype. We can transpose faster.
-                from pandas.core.arrays.masked import transpose_homogenous_masked_arrays
+                from pandas.core.arrays.masked import (
+                    transpose_homogeneous_masked_arrays,
+                )
 
                 if isinstance(self._mgr, ArrayManager):
                     masked_arrays = self._mgr.arrays
                 else:
                     masked_arrays = list(self._iter_column_arrays())
-                new_values = transpose_homogenous_masked_arrays(
+                new_values = transpose_homogeneous_masked_arrays(
                     cast(list[BaseMaskedArray], masked_arrays)
                 )
+            elif isinstance(dtypes[0], ArrowDtype):
+                # We have arrow EAs with the same dtype. We can transpose faster.
+                from pandas.core.arrays.arrow.array import (
+                    ArrowExtensionArray,
+                    transpose_homogeneous_arrow_extension_arrays,
+                )
+
+                if isinstance(self._mgr, ArrayManager):
+                    arrays = self._mgr.arrays
+                else:
+                    arrays = list(self._iter_column_arrays())
```

Comment:
cant do this unconditionally?

Reply:
Do you mean use `_iter_column_arrays` in both cases? The masked array implementation just above this uses the condition as well, presumably for performance:

```
import pandas as pd

pd.options.mode.data_manager = "array"

df = pd.DataFrame(columns=range(100000), dtype="int64[pyarrow]")

%timeit list(df._iter_column_arrays())
# 57.9 ms ± 2.7 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)

%timeit df._mgr.arrays
# 110 ns ± 8.79 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

Reply:
Fair enough, thanks for taking a look

Reply:
on second thought, I moved the condition into `DataFrame._iter_column_arrays` to avoid repeating it elsewhere